### ISSUE TITLE
Reusing WireMockExtension from abstract base class fails

### ISSUE THREAD

**Issue by timpeeters** [wiremock/wiremock#1647]
We are running into an issue with the JUnit 5 WireMockExtension. When the extension is defined as a static final on an abstract base class and re-used in several child classes. The WireMock server is stopped after the first test class finishes (@AfterAll) and not restarted when the second test class is executed (@BeforeAll).

The second test class will execute `beforeAll` and call `startServerIfRequired`. The `wireMockServer` field is already initialized by the first test class and skip initialization of the WireMock server. The test will then try to connect to a stopped server.

A solution could be to dereference the WireMock server in the `stopServerIfRunning` method. Causing the second test to re-initialize the WireMock server:

```
private void stopServerIfRunning() {
    if (wireMockServer != null && wireMockServer.isRunning()) {
        wireMockServer.stop();
    }
   wireMockServer = null;
}
```

WireMock version: 2.31.0
Reproducer: https://github.com/timpeeters/wiremock-bug

**Comment by tomakehurst** [comment]
I think this is already fixed in c4ece200f318a35a12f281cfc6bb2903800d904f

**Comment by timpeeters** [comment]
Fixed after upgrading to 2.32.0.
